docker/Dockerfile

```diff
@@ -32,3 +32,22 @@ RUN pip3 --no-cache-dir install --upgrade pip \
     && pip3 --no-cache-dir install flashinfer -i https://flashinfer.ai/whl/cu121/torch2.4/
 
 ENV DEBIAN_FRONTEND=interactive
+
+# build the wheel
+FROM base AS build
+
+RUN --mount=type=cache,target=/root/.cache/pip \
```

Comment:
I don't recommend using absolute paths. What's the purpose of adding them here?

Reply:
It is inspired from vllm's `Dockerfile`:
https://github.com/vllm-project/vllm/blob/main/Dockerfile#L47
```
RUN --mount=type=cache,target=/root/.cache/pip \
    python3 -m pip install -r requirements-cuda.txt
```

https://docs.docker.com/build/guide/mounts/#add-a-cache-mount
The default pip cache directory (as the docker build is running as root), is `/root/.cache/pip`.

The `--mount` in `Dockerfile` will mount the file (or volume) only during the `RUN` operation, and that will not preserved in the produced image.

This is kinda equivalent as:
- Run pip install with wheels cached to local machines (to deduplicate subsequent downloads in the future)
- Then remove the cache from the docker image layer.